src/Tizen.Multimedia.Vision/MediaVision/InferenceObjectDetector.cs

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using InteropOD = Interop.MediaVision.InferenceObjectDetection;
+
+namespace Tizen.Multimedia.Vision
+{
+    /// <summary>
+    /// Provides the ability to detect object.
+    /// </summary>
+    /// <feature>http://tizen.org/feature/vision.inference</feature>
+    /// <feature>http://tizen.org/feature/vision.inference.image</feature>
+    /// <since_tizen> 12 </since_tizen>
+    public class InferenceObjectDetector : IDisposable
+    {
+        private IntPtr _handle;
+        private bool _disposed;
+
+        /// <summary>Initializes a new instance of the <see cref="InferenceObjectDetector"/> class.</summary>
+        /// <exception cref="NotSupportedException">The required features are not supported.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public InferenceObjectDetector()
+        {
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.Inference);
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.InferenceImage);
+
+            InteropOD.Create(out _handle).Validate("Failed to create inference object detector.");
+
+            try
+            {
+                InteropOD.Configure(_handle).Validate("Failed to configure inference object detector.");
+                InteropOD.Prepare(_handle).Validate("Failed to prepare inference object detector.");
+            }
+            catch (Exception e)
+            {
+                Log.Error(MediaVisionLog.Tag, e.ToString());
+                InteropOD.Destroy(_handle);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the InferenceObjectDetector class.
+        /// </summary>
+        ~InferenceObjectDetector()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Detects object on the source image synchronously.
```

Comment:
if it finds many objects in the image, then please write plural: "Detects objects"

Reply:
I'll fix it.